SeeSharp/Geometry/PlyMesh.cs

```diff
@@ -306,21 +309,20 @@ private class PlyHeader {
         /// Will parse the header and populate the PlyHeader structure
         /// </summary>
         /// <param name="stream"></param>
-        /// <param name="header"></param>
-        /// <returns>List of errors</returns>
-        private static List<string> ParsePlyHeader(MixReader stream, ref PlyHeader header) {
-            List<string> errors = new();
+        /// <returns>PlyHeader or null if error</returns>
+        private static PlyHeader ParsePlyHeader(MixReader stream) {
+            PlyHeader header = new();
 
             bool eos = false;
             string magic = stream.ReadLineAsString(ref eos);
             if (magic != "ply") {
-                errors.Add("Trying to load invalid .ply file");
-                return errors;
+                Logger.Log($"Trying to load invalid .ply file '{stream.Path}'", Verbosity.Error);
+                return null;
```

Comment:
Errors like this one should probably throw an exception. Otherwise, this will just cause a NullReference exception further down the line which might be harder to debug than throwing an exception here.